typing/typedecl_unboxed.ml

```diff
@@ -34,20 +34,16 @@ let rec get_unboxed_type_representation env ty fuel =
         This Predef.type_int
     | {type_immediate = Always_on_64bits; _} ->
         Only_on_64_bits Predef.type_int
-    | {type_unboxed = {unboxed = false}} -> This ty
     | {type_params; type_kind =
-         Type_record ([{ld_type = ty2; _}], _)
-       | Type_variant [{cd_args = Cstr_tuple [ty2]; _}]
-       | Type_variant [{cd_args = Cstr_record [{ld_type = ty2; _}]; _}]}
-
+         Type_record ([{ld_type = ty2; _}], Record_unboxed _)
+       | Type_variant ([{cd_args = Cstr_tuple [ty2]; _}], Variant_unboxed)
+       | Type_variant ([{cd_args = Cstr_record [{ld_type = ty2; _}]; _}],
+                       Variant_unboxed)}
       ->
         let ty2 = match ty2.desc with Tpoly (t, _) -> t | _ -> ty2 in
         get_unboxed_type_representation env
           (Ctype.apply env type_params ty2 args) (fuel - 1)
-    | {type_kind=Type_abstract} -> Unavailable
-          (* This case can occur when checking a recursive unboxed type
-             declaration. *)
```

Comment:
Why did this case disappear? I suppose it refers to situations where we were not about to locate/expand the definition of a type abbreviation that is marked as unboxed?

Reply:
The case this code handles (a type which, while abstract, is somehow known to be unboxed) is no longer expressible. Abstract types generated while making recursive unboxed types no longer have any special tagging.

In these cases, the return value of this function changes from `Unavailable` to `This t`, where `t` is an abstract type. The only uses of the return value of this function are to compute its immediacy or compare with a builtin type, and in both of those cases `Unavailable` and `This <abstract>` are not distinguished.

(I think the interface to this function is poor, and the second patch in this series will change it)